### PR TITLE
Gets our food and drinks content up to date with /tg/, adds Boomer grill

### LINKED ISSUE #1674: Reading Cooking Desserts 101 removes ability to enter crafting menu.
## Reproduction: 
1. Buy Cooking Desserts 101 from Vending Machine (Dinner Vendor)
2. Read it until you learn how to cook new stuff.
3. Try to open crafting menu.

### PR DESCRIPTION
**Pull request by Cenrus**
## Related PRs

https://github.com/tgstation/tgstation/pull/44914
https://github.com/tgstation/tgstation/pull/45633
https://github.com/tgstation/tgstation/pull/46196
https://github.com/tgstation/tgstation/pull/46898
https://github.com/tgstation/tgstation/pull/48050
https://github.com/tgstation/tgstation/pull/46583
https://github.com/tgstation/tgstation/pull/45840
https://github.com/tgstation/tgstation/pull/44322
https://github.com/tgstation/tgstation/pull/49192
https://github.com/tgstation/tgstation/pull/46933
https://github.com/tgstation/tgstation/pull/50622
https://github.com/tgstation/tgstation/pull/46844
https://github.com/tgstation/tgstation/pull/45632
https://github.com/tgstation/tgstation/pull/50065

## Why it's good for the game

Better hamburger sprites, more things for the chef to make. Many thanks to /tg/ for these.

Closes #1674

:cl: Krysonism, trilbyspaceclone, Granpawalton, ArcaneMusic, Sneeker134, SteelSlayer, Fikou, Twaticus, bobubeu
add: More cooking content, ribs, cakes, new burgers.
imageadd: Burger resprites
add: Added new craftable Mob, the Butter Bear.
add: Five new plants!
add: Three new plant reagents!
tweak: Plants with high toxicity are now capable of taking pest damage.
balance: High pest levels are more damaging to plants.
imageadd: Added new plants sprites.
code: Coders can now make plant genes unextractable.
code: Adds the dunkable element. Converts items who were using the dunkable var to instead use this element and removes the dunkable var.
code: Adds a new "DUNKABLE" reagent flag for reagent containers. This determines whether or not the container can have items dunked into it.
add: Added lots of new donuts
add: 3 new dishes cooked with the new plants.
add: Added a nugget box. I'm lovin' it (tm)
del: You can no longer make cherry donuts.
tweak: Tea powder is less toxic and tastes better.
add: snowcones
imageadd: snowiest of snow cones
add: Adds the Grill, which can be powered by coal and wood, increases food nutritional value, taste, and adds grill marks.
add: Adds a new drink, Monkey Energy, works as fuel for the grill and increases your speed if you're a monkey, also adds a new reagent, Char, which works the magic in the grill.
tweak: Coal now turns into CO2 at high temperatures and is now a stack
add: Adds a strawberry ice cream sandwich sprite
add: Strawberry ice cream sandwich crafting recipe
add: Strawberry ice cream sandwich cargo bounty
/:cl:

### CODE REVIEW

**Comment by Cenrus** [comment]
PR is ready.

**Comment by qwertyquerty** [comment]
Testmerged onto sage

**Comment by Cenrus** [comment]
Fixed #1674

**Comment by PowerfulBacon** [comment]
Apparently Crafting menu cannot be opened after reading Cooking Desserts 101. It is blue(?|F) screen

**Comment by Cenrus** [comment]
Just pushed the fix, should take effect next round.

**Comment by Cenrus** [comment]
@qwertyquerty testmerge needs an update for the fix to take place

**Comment by Cenrus** [comment]
Wiki updated.

**Comment by qwertyquerty** [comment]
@Cenrus check if #1706 is related

**Comment by Cenrus** [comment]
@qwertyquerty 1706 isn't a testmerge issue, I just tested it, see the issue post for details.

**Comment by Cenrus** [comment]
Ready to merj

File: code/__DEFINES/reactions.dm [hunk 1]
```diff
@@ -33,7 +33,7 @@
 #define NOBLIUM_RESEARCH_AMOUNT				1000
 #define BZ_RESEARCH_SCALE					4
 #define BZ_RESEARCH_MAX_AMOUNT				400
-#define MIASMA_RESEARCH_AMOUNT				160
+#define MIASMA_RESEARCH_AMOUNT				40
```
Comment (Fikou) [inline code/__DEFINES/reactions.dm:36]: ?????

Comment (Cenrus) [reply code/__DEFINES/reactions.dm:36]: Was changed as a result of the corpse flower being added in 44322.

**Comment by Cenrus** [comment]
Crafting menu broke, one minute

**Comment by Cenrus** [comment]
Okay, now it's ready. this time for realsies.

**Pull request merged**